modules/sievefilters/modules.php

```diff
@@ -1344,3 +1344,62 @@ public function process() {
         }
     }
 }
+
+/**
+ * Check the status of an SIEVE server
+ * @subpackage sieve/handler
+ */
+class Hm_Handler_load_list_sieve_block extends Hm_Handler_Module {
+    public function process() {
+        list($success, $form) = $this->process_form(array('imap_server_id'));
+
+        if (!$success) {
+            return;
+        }
+
+        foreach ($this->user_config->get('imap_servers') as $idx => $mailbox) {
+            if ($idx == $this->request->post['imap_server_id']) {
+                $imap_account = $mailbox;
```

Comment:
Add `break;` after this line to exit the loop once the matching server ID is found.